src/steps/pentests.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  createDirectRelationship,
+  createIntegrationEntity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+  IntegrationMissingKeyError,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../client';
+import { IntegrationConfig } from '../types';
+import { VENDOR_ENTITY_KEY, SERVICE_ENTITY_KEY } from './account';
+
+export async function fetchPentests({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  const vendorEntity = await jobState.findEntity(VENDOR_ENTITY_KEY);
+  if (!vendorEntity) {
+    throw new IntegrationMissingKeyError(
+      `Expected Vendor with key to exist (key=${VENDOR_ENTITY_KEY})`,
+    );
+  }
+
+  const serviceEntity = await jobState.findEntity(SERVICE_ENTITY_KEY);
+  if (!serviceEntity) {
+    throw new IntegrationMissingKeyError(
+      `Expected Service with key to exist (key=${SERVICE_ENTITY_KEY})`,
+    );
+  }
+
+  await apiClient.iteratePentests(async (pentest) => {
+    const pentestProps = pentest.resource;
+    const assessmentEntity = await jobState.addEntity(
+      createIntegrationEntity({
+        entityData: {
+          source: pentest,
+          assign: {
+            _type: 'cobalt_pentest',
+            _class: 'Assessment',
+            _key: pentestProps.id,
+            name: pentestProps.title,
+            displayName: pentestProps.title,
+            category: 'Penetration Testing',
+            summary: pentestProps.title, //required prop for Assessment
+            internal: false,
```

Comment:
Do we know for a fact that this is not `internal`?

@JupiterOne/integrations I think we need to consider allowing these kinds of required properties to be specified as `null` when the value is indeterminate. Otherwise, we're making an unsubstantiated truth claim which may lead to unintended activity or prevent activity that needs to occur.

Reply:
How would it be `internal` if Cobalt ran the pentest? Maybe I'm misunderstanding the use of the term here. I was thinking "internal" meant the client themselves ran the pentest using their own people, and it would be false if the pentest was outsourced. What should the term mean here?